### DISCUSSION TITLE
Help required regarding error displaying over Crashlytics dashboard page

### DISCUSSION THREAD

**Discussion by Yandamuri** [invertase/react-native-firebase#6649] [Q&A] [answered]
I see following error message when Crashlytics dashboard page is opened.

> On Android, we're aware of an issue in the previous version of the Crashlytics SDK. We strongly recommend updating to the SDK version listed in the GitHub issue.

And this github issue comment link was given for help --> https://github.com/firebase/firebase-android-sdk/issues/4223#issuecomment-1284447351

This comment apparently suggests following thing

> The Firebase BoM has been updated with the fix, in version 31.0.1.

But I do not use BOM and I am working on `react-native` project. This is the guide I followed to configure `Crashlytics` --> https://invertase.io/blog/react-native-firebase-crashlytics-configuration

There wasn't any discussion about installing `BOM`.

1. Do i need to take any action to get rid of mentioned error?
2. Is this error going to affect `Crashlytics` service of my app?

**Comment by mikehardy** [comment, marked as answer]
If you use `@react-native-firebase/app` (which you must use, to use any other react-native-firebase packages) then we use the BoM and manage the versions for you, so you are using it but you do not have to think about it normally.

For Android, if you are on current versions here then you already have the necessary fixes:

https://github.com/invertase/react-native-firebase/blob/d654d00e06f7d638828653cfaa64ef9ad5e343d9/packages/app/package.json#L76

We are on firebase-android-sdk 31.0.2 now.

You may also consult our changelog any time to see what changes are, and that would have answered your question:

https://invertase.io/blog/react-native-firebase-versioning

There were only two days where BoM 31.0.0 was in place here before 31.0.1 was released and I released here to get it out to users:
https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#1611-2022-10-21

**Comment by Yandamuri** [comment]
@mikehardy Thank you for your prompt response.

Now, I agree that no need add BOM explicitly as it is already part of `@react-native-firebase/app`.  And I have gone through the changelog, but i didn't find any relevant info to get rid of the error.

Is there anything that I can do to get get rid of this error?

BTW, I am `@react-native-firebase/app": "11.5.0` and  `@react-native-firebase/crashlytics": "11.5.0`

**Reply by mikehardy** [reply]
It's a warning, not an error. It was a serious enough issue they made it a banner but that also means you cannot dismiss it.

 I linked the changelog where I noted BoM changes, so you can see where we increased the version to the one that has the fix. That was the purpose of the changelog link I posted for you.

If you are on current versions here you have no actions to take.

**Reply by mikehardy** [reply]
Your versions (now that I look) are unsupported and too old to be considered valid any more. You should plan time to upgrade, noting the breaking changes described in the changelog entries for major versions.

**Comment by Yandamuri** [comment]
@mikehardy 

This is my `Firebase` version,

```
 /node_modules/@react-native-firebase/app/package.json
"firebase": "26.8.0"
```

This could be the reason why I am seeing the warning.

I think I found the root cause of the issue. Following are the `Firebase` packages that I use,

```
    @react-native-firebase/app: "11.5.0",
    @react-native-firebase/crashlytics: "11.5.0",
    @react-native-firebase/iid: "11.5.0",
    @react-native-firebase/messaging: "11.5.0",
```

The latest version of `@react-native-firebase/iid` is `11.5.0` and it depends on `@react-native-firebase/app: 11.5.0` . If i increase the version of `@react-native-firebase/app` to the latest, Then I am getting following warning while installing `pods`.

> [!] NPM package '@react-native-firebase/iid' depends on '@react-native-firebase/app' v11.5.0 but found v16.4.1, this might cause build issues or runtime crashes.

<img width="574" alt="Screenshot 2022-11-03 at 11 01 59 AM" src="https://user-images.githubusercontent.com/31983789/199653531-f7c935c8-dccc-4803-a4a2-4bc61439c578.png">

IF I have to consider the above warning, Then I can never maintain the latest version of ` @react-native-firebase/app`. Because  `@react-native-firebase/iid`'s version is always lower than `@react-native-firebase/app`.

I believe the above `pods` warning is thrown by `@react-native-firebase`. In that case,

May I know How to always maintain the latest version of ` @react-native-firebase/app` and still making it compatible with ` @react-native-firebase/iid`?

**Reply by mikehardy** [reply]
Iid is removed as a package, installations took over it's role. Remove iid and stop using it, upgrade to current, you'll be fine
